### PR TITLE
Adding algorithm "Does scheme-part match another scheme-part?"

### PR DESCRIPTION
**Pull request by Sun77789**

### CODE REVIEW

**Comment by Sun77789** [comment]
@mikewest ready for review!

File: index.src.html [hunk 1]
```diff
@@ -3480,6 +3467,30 @@ <h5 id="match-url-to-source-expression" algorithm>
 
   5.  Return "`Does Not Match`".
 
+  <h5 id="match-two-scheme-parts" algorithm>
+    Does |scheme-part| match another |scheme-part|?
+  </h5>
+
+  Given <a grammar>`scheme-part`</a> A and <a grammar>`scheme-part`</a> B, this 
+  algorithm returns "`Matches`" if A matches B, and returns "`Does Not Match`" 
+  otherwise.
```
Comment (mikewest) [inline index.src.html]: You don't really want these both to be `scheme-part`s, as you're passing in a URL's scheme as the second argument in step 2.1 above. I think it's probably reasonable to make this algorithm operate on [ASCII strings](https://url.spec.whatwg.org/#ascii-string) instead.

Comment (mikewest) [inline index.src.html]: It's probably also worth adding a note that this algorithm is not transitive. That is, A matching B does not mean that B will match A.

File: index.src.html [hunk 2]
```diff
@@ -3480,6 +3467,30 @@ <h5 id="match-url-to-source-expression" algorithm>
 
   5.  Return "`Does Not Match`".
 
+  <h5 id="match-two-scheme-parts" algorithm>
```
Comment (mikewest) [inline index.src.html]: I'd make this something like `id="match-schemes"`. You can follow it up with `match-hosts`, `match-ports`, etc. No need to have long and descriptive IDs. :)

File: index.src.html [hunk 3]
```diff
@@ -3467,13 +3467,15 @@ <h5 id="match-url-to-source-expression" algorithm>
...
 
-  <h5 id="match-two-scheme-parts" algorithm>
+  <h5 id="match-schemes" algorithm>
     Does |scheme-part| match another |scheme-part|?
   </h5>
 
-  Given <a grammar>`scheme-part`</a> A and <a grammar>`scheme-part`</a> B, this 
-  algorithm returns "`Matches`" if A matches B, and returns "`Does Not Match`" 
-  otherwise.
+  Given two ASCII strings A and B, this algorithm returns "`Matches`" if A
```
Comment (mikewest) [inline index.src.html]: Nit: can you link this? `<a>ASCII strings</a>` should work.

**Comment by Sun77789** [comment]
Marked as non-substantive for IPR from ash-nazg.

**Comment by Sun77789** [comment]
I had to mark it as non-substantive as otherwise, lpr check was failing.

**Comment by mikewest** [comment]
@wseltzer: What do I need to do to have @Sun77789 added as a contributor? She's working with me until the end of December.

@Sun77789: LGTM!

**Comment by wseltzer** [comment]
@mikewest can @Sun77789 join the WG? and thanks!

**Comment by wseltzer** [comment]
OK to merge. Contribution from a Google intern is covered by Google's WG participation.

**Pull request merged**